sentinel-dashboard/src/main/java/com/alibaba/csp/sentinel/dashboard/controller/AuthorityRuleController.java

```diff
@@ -69,7 +69,9 @@
                                                                         @RequestParam String ip,
                                                                         @RequestParam Integer port) {
         AuthUser authUser = authService.getAuthUser(request);
-        authUser.authTarget(app, PrivilegeType.READ_RULE);
+        if (!authUser.authTarget(app, PrivilegeType.READ_RULE)) {
```

Comment:
Writing the duplicate boilerplate code in every controller handler is tedious :(
Maybe we need a unified, simplified filter or other mechanisms to achieve this. See #745 for more discussions.

Reply:
Yes, I try to improve it

Reply:
> Writing the duplicate boilerplate code in every controller handler is tedious :(
> Maybe we need a unified, simplified filter or other mechanisms to achieve this. See #745 for more discussions.

I have pushed new code.
Use AuthInterceptor to intercept the AuthAction annotation
`
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Target({ElementType.METHOD})
public @interface AuthAction {

    AuthService.PrivilegeType value();

    String targetName() default "app";

    String message() default "No privilege";
}
`
Value is the privilege.
TargetName is the auth target, from the request parameter(post or get), usually app or ip, default app.
Message is the response message when do action without privilege.